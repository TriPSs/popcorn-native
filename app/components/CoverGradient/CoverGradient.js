import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

import colors from 'modules/colors'

const styles = StyleSheet.create({

  linearGradient: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  },

})

export const CoverGradient = ({ start, end }) => (
  <LinearGradient
    start={start}
    end={end}
    colors={['transparent', colors.BACKGROUND]}
    locations={[0, 1]}
    style={styles.linearGradient} />
)

CoverGradient.propTypes = {
  start: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
  }),

  end: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
  }),
}

CoverGradient.defaultProps = {
  start: {
    x: 0,
    y: 0.1,
  },

  end: {
    x: 0,
    y: 1,
  },
}

export default CoverGradient
