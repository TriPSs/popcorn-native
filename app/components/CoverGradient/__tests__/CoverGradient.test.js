import React from 'react'
import { shallow } from 'enzyme'

import CoverGradient from '../CoverGradient'

describe('(Component) CoverGradient', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <CoverGradient />,
    )

    expect(result).toMatchSnapshot()
  })

})
