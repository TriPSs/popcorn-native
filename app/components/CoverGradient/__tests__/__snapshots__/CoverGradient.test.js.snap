// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CoverGradient should render with defaults 1`] = `
<LinearGradient
  colors={
    Array [
      "transparent",
      "#202125",
    ]
  }
  end={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  start={
    Object {
      "x": 0,
      "y": 0.1,
    }
  }
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
